### ISSUE TITLE
Lightbomb deals damage 1 at a time, causing Frothing Berserker to buff and take more damage mid-spell

### ISSUE THREAD

**Issue by Patashu** [HearthSim/hs-bugs#133]
When casting Lightbomb and there is a Frothing Berserker on the board, the Frothing Berserker will activate as other minions are damaged by Lightbomb and may grow large enough in Attack to subsequently get killed by Lightbomb. This is unintended as all damage should be dealt before any damage triggers are activated.
* http://www.reddit.com/r/hearthstone/comments/2zyed2/til_playing_lightbomb_while_you_have_a_frothing/
* https://twitter.com/bdbrode/status/580110768131051520
* https://www.youtube.com/watch?v=7narpEF6bUY
* https://www.youtube.com/watch?v=GWWdMCrN-J4&t=1m00s
* https://www.youtube.com/watch?v=vXUTXbeVGYc&feature=youtu.be&t=392

However, in the middle of a [[Lightbomb]] cast, auras do not turn off. For example, a [[Dire Wolf Alpha]] being reduced to 0 HP will not allow a [[Mechwarper]] next to it to survive - it still takes 3 damage and dies.
* https://www.youtube.com/watch?v=DcpOyH4weFg

**Comment by Patashu** [comment]
Ported this bug over because of a report that's in Patch 3.0: https://www.youtube.com/watch?v=vXUTXbeVGYc&feature=youtu.be&t=392

**Comment by culinko** [comment]
In the Frothing case, if he was played first, he will deal only 2 damage to himself.

**Comment by Patashu** [comment]
Right, Frothing has to be played after minions that take damage from Lightbomb, in order of play, for this bug to happen.

**Comment by Patashu** [comment]
Confirmed in 3.2:

http://www.twitch.tv/playhearthstone/v/24516745

Go to approximately 1:18:30 in this VOD. Lightbomb is used against a frothing board and the damage is taken one at a time.

**Comment by culinko** [comment]
Well, this video only shows the visual bug (#11). And because frothing was played first, we don't know if it would take the additional damage. But I doubt they changed it recently.

**Comment by culinko** [comment]
I tested Frothing as a 3rd minion and as a 4th minion and Lightbomb did 3 damage in both cases. I had no spellpower and Frothing had bases stats before casting the Lightbomb. Attaching a log and screenshot of the second test (Frothing playing as 3rd minion):
* https://gist.github.com/culinko/0cb8a6edfd303378730d
*  http://imgur.com/txHcfeP

**Comment by theAprel** [comment]
Frothing no longer buffs mid-Lightbomb:

http://www.twitch.tv/nl_kripp/v/29317986?t=2h2m30s

(observed in 4.1.0.10956)

**Comment by jleclanche** [comment]
Hm the animation is still bugged, which is really weird because it means the damage triggers still happen one by one; lightbomb just snapshots the initial damage.

Still, we can consider it fixed now.

**Comment by Patashu** [comment]
Well hang on - we can't close it yet, until we test it again with a Frothing Berserker that starts at 2/4 base stats and it takes 2 damage after earlier minions take damage.

**Comment by culinko** [comment]
@jleclanche my post with the screesnhot is from 4.1. Please, don't consider this closed until we're sure it no longer happens.

**Comment by jleclanche** [comment]
ok, I saw it on milestone 3.2 so I just assumed it was an old post.

**Comment by Patashu** [comment]
Suggestion: As of Patch 4.0 or 4.1, self-buffs increasing in their stack no longer immediately update Health/Attack, but self-buffs first attachment does immediately update Health/Attack, so the Frothing Berserker/Lightbomb bug exists only to add 1 damage to unbuffed Frothings.

That would mean this bug is now related to https://github.com/HearthSim/hs-bugs/issues/325

**Comment by Patashu** [comment]
https://www.youtube.com/watch?v=R9dnoreiVwM&feature=youtu.be

https://www.youtube.com/watch?v=COrdQhWmUiE&feature=youtu.be

Confirmed in 4.2

**Comment by Patashu** [comment]
Confirmed in 4.3

https://www.reddit.com/r/hearthstone/comments/4f64wn/frothing_at_2_atk_lightbomb_frothing_deals_itself/

**Comment by PattuX** [comment]
Nothing changed, buffed Frothing works as intended, unbuffed takes 3.

**Comment by PattuX** [comment]
An enrage minion getting damaged within the spell text phase of Lightbomb (via Imp Gang Boss + Knije Juggler) gets damaged by its base attack, not the buffed value. That means, enrage does not apply instantly, but probably in the next Aura Update (A/H).

**Comment by culinko** [comment]
Still happens in 7.0.

**Comment by Patashu** [comment]
still happens w/ Elemental Destruction

https://www.youtube.com/watch?v=KFXkwbwCf-A&feature=youtu.be#t=6m15s

**Comment by Jetz72** [comment]
Still affects Lightning Storm as well. If the spell is given Lifesteal via Omega Mind, the healing won't be properly consolidated.
https://us.forums.blizzard.com/en/hearthstone/t/omega-mind-doesnt-work-with-lightning-storm/9010/4
>My bad, it works well. I re tried 2 times. But the animation of life steal does not display correctly : it displays “+2” or “+3” instead of the real amount of life gained.
>
>I recorded it : https://youtu.be/Se9iccC2_Zo
Here, I had 21 points of life and I gained 7 points => 28 points. But it displayed only 2.
